app/src/main/java/ca/dal/cs/athletemonitor/athletemonitor/RecordActivity.java

```diff
@@ -297,16 +347,16 @@ private static void saveToFirebase(String dbRef, long time, List<Location> locat
     }
 
     private boolean checkForLocPermission() {
-        if (ContextCompat.checkSelfPermission(this, android.Manifest.permission.ACCESS_FINE_LOCATION)
-                != PackageManager.PERMISSION_GRANTED) {
-            return false;
+            if (ContextCompat.checkSelfPermission(this, android.Manifest.permission.ACCESS_FINE_LOCATION)
+                    != PackageManager.PERMISSION_GRANTED) {
+                return false;
+            }
+            else
+                return true;
```

Comment:
simply:
`return ContextCompat.checkSelfPermission(this, android.Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED`